node/node.go

```diff
@@ -1337,7 +1458,15 @@ func (n *Node) ConfigureRPC() error {
 		Logger: n.Logger.With("module", "rpc"),
 
 		Config: *n.config.RPC,
-	})
+	}
+	if n.config.Mode == cfg.ModeValidator {
+		pubKey, err := n.privValidator.GetPubKey(context.TODO())
+		if pubKey == nil || err != nil {
+			return fmt.Errorf("can't get pubkey: %w", err)
```

Comment:
is this error salient when pubkey is nil and error is nil?

Reply:
yup. I'm not sure if there is a case where it happens, but if it does we should still error